.github/workflows/images.yaml

```diff
@@ -48,4 +48,7 @@ jobs:
             platform=${{ matrix.platform }}
             BUILDKIT_INLINE_CACHE=1
           push: true
-          cache-from: type=registry,ref=nubots/nubots:${{ matrix.platform }}
+          cache-from: |
+            type=registry,ref=nubots/nubots:pull-request-${{ github.event.number }}
+            type=registry,ref=nubots/nubots:generic
```

Comment:
You should be caching from the type of this image you're building, not generic all the time. The nuc7i7bnh won't have much in common with the generic image, so caching is pointless. Cache from the correct image